sormas-backend/src/test/java/de/symeda/sormas/backend/deletionconfiguration/CoreEntityDeletionServiceTest.java

```diff
@@ -172,7 +176,64 @@ public void testCasePermanentDeletion() throws IOException {
 		assertNull(getTravelEntryFacade().getByUuid(travelEntry.getUuid()).getResultingCase());
 		assertNull(getImmunizationFacade().getByUuid(immunization.getUuid()).getRelatedCase());
 	}
-	
+
+	@Test
+	public void testSamplePermanentDeletion() {
+
+		TestDataCreator.RDCF rdcf = creator.createRDCF();
+		UserDto user = creator.createUser(rdcf, UserRole.ADMIN, UserRole.NATIONAL_USER);
+		PersonDto person = creator.createPerson();
+		CaseDataDto caze = creator.createCase(user.toReference(), person.toReference(), rdcf);
+		SampleDto sample = creator.createSample(caze.toReference(), user.toReference(), rdcf.facility);
+		SampleDto referralSample =
+			creator.createSample(caze.toReference(), user.toReference(), rdcf.facility, s -> s.setReferredTo(sample.toReference()));
+		creator.createPathogenTest(sample.toReference(), caze);
+		creator.createAdditionalTest(sample.toReference());
+		LabMessageDto labMessage = creator.createLabMessage(lm -> lm.setSample(sample.toReference()));
+
+		getSampleFacade().deleteSample(sample.toReference());
+
+		List<PathogenTest> pathogenTests = getPathogenTestService().getAll();
+		assertEquals(2, getSampleService().count());
+		assertTrue(getSampleService().getByUuid(sample.getUuid()).isDeleted());
+		assertEquals(1, pathogenTests.size());
+		assertTrue(pathogenTests.get(0).isDeleted());
+		assertEquals(1, getAdditionalTestService().count());
+		assertNull(getSampleService().getByUuid(referralSample.getUuid()).getReferredTo());
+		assertNull(getLabMessageService().getByUuid(labMessage.getUuid()).getSample());
+
+		useSystemUser();
+		getCoreEntityDeletionService().executePermanentDeletion();
+		loginWith(user);
+
+		assertEquals(1, getSampleService().count());
+		assertEquals(0, getPathogenTestService().count());
+		assertEquals(0, getAdditionalTestService().count());
+		assertEquals(1, getLabMessageService().count());
```

Comment:
why isn't the lab message removed here ?

Reply:
@BarnaBartha Because there was no specification that lab messages should be deleted when their associated samples are deleted :)